packages/wxt/src/client/content-scripts/ui/index.ts

```diff
@@ -67,16 +97,38 @@ export function createIframeUi<TMounted>(
   wrapper.appendChild(iframe);
 
   let mounted: TMounted | undefined = undefined;
+  let autoMountInstance: AutoMount | undefined = undefined;
   const mount = () => {
     applyPosition(wrapper, iframe, options);
     mountUi(wrapper, options);
     mounted = options.onMount?.(wrapper, iframe);
   };
+  const autoMount = (autoMountOptions?: AutoMountOptions) => {
+    if (autoMountInstance) {
+      throw Error('autoMount is already set.');
+    }
+    autoMountInstance = autoMountUi(
+      { mount, unmount, stopAutoMount },
+      {
+        ...options,
+        ...autoMountOptions,
+      },
+    );
+    return stopAutoMount;
+  };
```

Comment:
There's a bit of code duplication here, doing the same thing for each of the three UI functions... Would it be better if `autoMount` was a separate utility function exported from `wxt/client` rather than an instance function?

```diff
- ui.autoMount()
+ autoMount(ui)
```

That would make it easier to reduce code duplication... Not sure if that'd be better than/easier to use or see compared to what you're currently doing, just throwing out an idea.

_Edit: the anchor is private to the instance, so this probably wouldn't be possible without exposing it... Hmm, is there a better way to reduce code duplication? Maybe a function like this?_

```ts
interface AutoMountable {
  mount(): void;
  remove(): void;
}

interface AutoMount {
   autoMount(...): StopAutoMount
}

function addAutoMount<T extends AutoMountable>(
  anchor: ...,
  ui: T,
): T & AutoMount {
  ...
}
```

_These interface names probably conflict with what you've already defined, just using them as an example. They might already exist in your branch too, can't remember._

Reply:
Yeah...if we needed a fourth type of built-in UI, I think it would be tough. Let's see if your helper function pattern will work. But, it is necessary to let the user set the anchor twice.... 🤔

Reply:
Yeah, I would rather not have users pass the anchor twice.